.github/workflows/build.yml

```diff
@@ -0,0 +1,21 @@
+# This workflow builds the repo
+
+name: Build
+
+# Triggers the workflow on push or pull request events but only for the main branch
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main, release* ]
+  workflow_dispatch:
+
+jobs:
+  build-ubuntu:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: sudo apt update && sudo apt install -y libacl1-dev libncurses5-dev
+      - name: Build
+        run: cmake -B build && cmake --build build && ./build/dds_gateway_moduletests
```

Comment:
The build is broken since the FetchContent does not build iceoryx with tests enabled:

```
INFO Using CycloneDDS stack
CMake Error at test/CMakeLists.txt:20 (find_package):
  By not providing "Findiceoryx_hoofs_testing.cmake" in CMAKE_MODULE_PATH
  this project has asked CMake to find a package configuration file provided
  by "iceoryx_hoofs_testing", but CMake did not find one.

  Could not find a package configuration file provided by
  "iceoryx_hoofs_testing" with any of the following names:

    iceoryx_hoofs_testingConfig.cmake
    iceoryx_hoofs_testing-config.cmake

  Add the installation prefix of "iceoryx_hoofs_testing" to CMAKE_PREFIX_PATH
  or set "iceoryx_hoofs_testing_DIR" to a directory containing one of the
  above files.  If "iceoryx_hoofs_testing" provides a separate development
  package or SDK, be sure it has been installed.


-- Configuring incomplete, errors occurred!
```

Reply:
We need to call `find_package(iceoryx_hoofs_testing)`, etc. and maybe also pass `-DBUILD_TEST`. I tried it once but I'm not quite sure anymore what exactly was needed.

Reply:
Fixed with 49ba03d66a1743804d14e1454677cd48b651e816